source/SongCore/HarmonyPatches/BeatmapVersionDetectionPatch.cs

```diff
@@ -1,35 +1,24 @@
 ﻿using System;
-using System.Text.RegularExpressions;
 using HarmonyLib;
 using JetBrains.Annotations;
+using UnityEngine;
 
 namespace SongCore.HarmonyPatches
 {
-    // This patch fixes the base game implementation, which fails with maps that either have whitespace in their difficulty files
-    // or have no version declared. Without this patch affected maps don't load when selected.
+    // This patch fixes the base game implementation, which fails with maps that have no version declared.
+    // Without this patch affected maps don't load when selected.
     [HarmonyPatch(typeof(BeatmapSaveDataHelpers), nameof(BeatmapSaveDataHelpers.GetVersion))]
     [UsedImplicitly]
     internal static class BeatmapVersionDetectionPatch
     {
-        private static readonly Regex VersionRegex = new Regex(
-            @"""_?version""\s*:\s*""(?<version>[0-9]+\.[0-9]+\.?[0-9]?)""",
-            RegexOptions.Compiled | RegexOptions.CultureInvariant
-        );
-
         private static readonly Version FallbackVersion = new Version("2.0.0");
 
         [UsedImplicitly]
         private static bool Prefix(string data, ref Version __result)
         {
-            var truncatedText = data.Substring(0, 50);
-            var match = VersionRegex.Match(truncatedText);
-            if (!match.Success)
-            {
-                __result = FallbackVersion;
-                return false;
-            }
+            string? version = JsonUtility.FromJson<BeatmapSaveDataHelpers.VersionSerializedData>(data).v;
+            __result = version != null ? new Version(version) : FallbackVersion;
 
-            __result = new Version(match.Groups["version"].Value);
             return false;
         }
```

Comment:
Regardless of my previous comment, I think it might be cleaner to just use a postfix here instead, that way... the initial logic doesn't need to be duplicated.

Reply:
Base game implementation is throwing when accessing the `Version` constructor due to `v` being null.

Reply:
My bad, completely overlooked that fact and somehow thought it would just return `null`